Changelog.md

```diff
@@ -12,6 +12,7 @@ Compiler Features:
  * Metadata: Update the swarm hash, changes ``bzzr0`` to ``bzzr1`` and urls to use ``bzz-raw://``.
  * Standard JSON Interface: Compile only selected sources and contracts.
  * Standard JSON Interface: Provide secondary error locations (e.g. the source position of other conflicting declarations).
+ * Yul Optimizer: Do not inline function if high cost expression will be duplicated.
```

Comment:
Could this be made more precise? something like `Inline a function if it's either called at most once or contains an expression whose cost is at most one` OR `Do not inline functions that are called more than once and contain an expression whose cost is greater than one`?

Reply:
What I have done is to make sure that the following code won't trigger expression inliner (because if inlined, it will need to do 3 mload instead of 1 mload):
```
function ref3(a) -> x { x := add(a, mul(a, a)) }
ref3(mload(42))
```
(So probably not relevant to how many times the function is being called)

Reply:
I see. I still feel high cost could be made explicit with something like `more than one`. High cost feels exaggerated for anything greater than one :-)

Reply:
Actually, the reason I was a bit confused here was that in the above snippet

```
function ref3(a) -> x { x := add(a, mul(a, a)) }
ref3(mload(42))
```

I assumed the cost you are referring to is the cost of the expression inside the function body and not the inlined expression. Just realized what I assumed does not make any sense because that is not the one getting duplicated :-)

Reply:
> I see. I still feel high cost could be made explicit with something like `more than one`. High cost feels exaggerated for anything greater than one :-)

Yea I guess I can change `high cost` to `non-cheap`